tree/ntuple/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -65,6 +67,129 @@ public:
    std::unique_ptr<ROOT::Internal::RPageSource> CreatePageSource() const;
 };
 
+class RNTupleProcessorEntry {
+   friend class RNTupleSingleProcessor;
+   friend class RNTupleChainProcessor;
+   friend class RNTupleJoinProcessor;
+   template <typename T>
+   friend class RNTupleProcessorValuePtr;
+
+private:
+   std::unique_ptr<ROOT::REntry> fEntry;
+   bool fIsValid = true;
+
+   RNTupleProcessorEntry(std::unique_ptr<ROOT::REntry> entry) : fEntry(std::move(entry)) {}
+
+   void SetValid(bool valid) { fIsValid = valid; }
```

Comment:
Just as a heads-up, down the line we may need to add more semantic information to whether the entry is valid or not, see e.g. https://github.com/root-project/root/blob/b4955fc7fb18b15ee2ab96358571a9de1a0e495b/tree/treeplayer/inc/TTreeReader.h#L149-L162